crates/sdk/src/sys/mock.rs

```diff
@@ -0,0 +1,62 @@
+use super::types::*;
+
+#[no_mangle]
+extern "C" fn panic(_loc: Location) -> ! {
+    panic!("Although your build was successful, it doesn't mean you're allowed to call this function in a non WASM target!")
```

Comment:
can you define a declarative macro to reduce duplication?

Reply:
Something like `The function_name is supposed to call only on WASM target. Panicking!`?